inspirehep/modules/records/receivers.py

```diff
@@ -206,3 +214,27 @@ def enhance_after_index(sender, json, record, *args, **kwargs):
 
     elif is_data(json):
         populate_citations_count(record=record, json=json)
+
+
+@after_record_insert.connect
+@after_record_update.connect
+def index_new_cited_records_after_record_update(sender, record, *args, **kwargs):
+    """Index records whose reference has been added or deleted"""
+    pids = record.get_modified_references()
+
+    if not pids:
+        return
+
+    uuids = [
+        pid[0] for pid in
+        db.session.query(PersistentIdentifier.object_uuid).filter(
+            PersistentIdentifier.object_type == 'rec',
+            tuple_(PersistentIdentifier.pid_type, PersistentIdentifier.pid_value).in_(pids)
+        ).all()
```

Comment:
minor: I think results of queries are iterable, so you probably don't need `.all()` here.